src/plugins/vis_type_xy/public/to_ast.ts

```diff
@@ -62,15 +171,13 @@ export const toExpressionAst: VisToExpressionAst<VisParams> = async (vis, params
     }
   }
 
-  const visConfig = { ...vis.params };
```

Comment:
This was to prevent an infinite update loop, particularly with date histogram charts. I think this should be fixed with these changes but could you play around with the date histogram visualization just updating the config and see if it hangs. Thanks!

Reply:
I haven't find any problems with date histograms now. I guess you a re right and my PR fixed it.